### PR TITLE
Add relevant hyperlinks to the documentation

### LINKED ISSUE #31: The libz blitz API guidelines checklist
# Rust API Guidelines Checklist

- **Organization** *(crate is structured in an intelligible way)*
  - [x] Crate root re-exports common functionality ([C-REEXPORT])
  - [x] Modules provide a sensible API hierarchy ([C-HIERARCHY])
- **Naming** *(crate aligns with Rust naming conventions)*
  - [x] Casing conforms to RFC 430 ([C-CASE])
  - [x] Ad-hoc conversions follow `as_`, `to_`, `into_` conventions ([C-CONV])
  - [x] Methods on collections that produce iterators follow `iter`, `iter_mut`, `into_iter` ([C-ITER])
  - [x] Iterator type names match the methods that produce them ([C-ITER-TY])
  - [x] Ownership suffixes use `_mut` and `_ref` ([C-OWN-SUFFIX])
  - [x] Single-element containers implement appropriate getters ([C-GETTERS])
  - [x] Feature names are free of placeholder words ([C-FEATURE])
- **Interoperability** *(crate interacts nicely with other library functionality)*
  - [x] Types eagerly implement common traits ([C-COMMON-TRAITS])
    - `Copy`, `Clone`, `Eq`, `PartialEq`, `Ord`, `PartialOrd`, `Hash`, `Debug`,
      `Display`, `Default`
  - [x] Conversions use the standard traits `From`, `AsRef`, `AsMut` ([C-CONV-TRAITS])
  - [x] Collections implement `FromIterator` and `Extend` ([C-COLLECT])
  - [x] Data structures implement Serde's `Serialize`, `Deserialize` ([C-SERDE])
  - [x] Crate has a `"serde"` cfg option that enables Serde ([C-SERDE-CFG])
  - [x] Types are `Send` and `Sync` where possible ([C-SEND-SYNC])
  - [x] Error types are `Send` and `Sync` ([C-SEND-SYNC-ERR])
  - [x] Error types are meaningful, not `()` ([C-MEANINGFUL-ERR])
  - [x] Binary number types provide `Hex`, `Octal`, `Binary` formatting ([C-NUM-FMT])
- **Macros** *(crate presents well-behaved macros)*
  - [x] Input syntax is evocative of the output ([C-EVOCATIVE])
  - [x] Macros compose well with attributes ([C-MACRO-ATTR])
  - [x] Item macros work anywhere that items are allowed ([C-ANYWHERE])
  - [x] Item macros support visibility specifiers ([C-MACRO-VIS])
  - [x] Type fragments are flexible ([C-MACRO-TY])
- **Documentation** *(crate is abundantly documented)*
  - [x] Crate level docs are thorough and include examples ([C-CRATE-DOC])
  - [x] All items have a rustdoc example ([C-EXAMPLE])
  - [x] Examples use `?`, not `try!`, not `unwrap` ([C-QUESTION-MARK])
  - [x] Function docs include error conditions in "Errors" section ([C-ERROR-DOC])
  - [x] Function docs include panic conditions in "Panics" section ([C-PANIC-DOC])
  - [ ] Prose contains hyperlinks to relevant things ([C-LINK])
  - [x] Cargo.toml includes all common metadata ([C-METADATA])
    - authors, description, license, homepage, documentation, repository,
      readme, keywords, categories
  - [x] Cargo.toml documentation key points to "https://docs.rs/CRATE" ([C-DOCS-RS])
  - [x] Crate sets html_root_url attribute "https://docs.rs/CRATE/VER.SI.ON" ([C-HTML-ROOT])
  - [x] Release notes document all significant changes ([C-RELNOTES])
- **Predictability** *(crate enables legible code that acts how it looks)*
  - [x] Smart pointers do not add inherent methods ([C-SMART-PTR])
  - [x] Conversions live on the most specific type involved ([C-CONV-SPECIFIC])
  - [x] Functions with a clear receiver are methods ([C-METHOD])
  - [x] Functions do not take out-parameters ([C-NO-OUT])
  - [x] Operator overloads are unsurprising ([C-OVERLOAD])
  - [x] Only smart pointers implement `Deref` and `DerefMut` ([C-DEREF])
  - [x] `Deref` and `DerefMut` never fail ([C-DEREF-FAIL])
  - [x] Constructors are static, inherent methods ([C-CTOR])
- **Flexibility** *(crate supports diverse real-world use cases)*
  - [x] Functions expose intermediate results to avoid duplicate work ([C-INTERMEDIATE])
  - [x] Caller decides where to copy and place data ([C-CALLER-CONTROL])
  - [x] Functions minimize assumptions about parameters by using generics ([C-GENERIC])
  - [x] Traits are object-safe if they may be useful as a trait object ([C-OBJECT])
- **Type safety** *(crate leverages the type system effectively)*
  - [x] Newtypes provide static distinctions ([C-NEWTYPE])
  - [x] Arguments convey meaning through types, not `bool` or `Option` ([C-CUSTOM-TYPE])
  - [x] Types for a set of flags are `bitflags`, not enums ([C-BITFLAG])
  - [x] Builders enable construction of complex values ([C-BUILDER])
- **Dependability** *(crate is unlikely to do the wrong thing)*
  - [x] Functions validate their arguments ([C-VALIDATE])
  - [x] Destructors never fail ([C-DTOR-FAIL])
  - [x] Destructors that may block have alternatives ([C-DTOR-BLOCK])
- **Debuggability** *(crate is conducive to easy debugging)*
  - [x] All public types implement `Debug` ([C-DEBUG])
  - [x] `Debug` representation is never empty ([C-DEBUG-NONEMPTY])
- **Future proofing** *(crate is free to improve without breaking users' code)*
  - [x] Structs have private fields ([C-STRUCT-PRIVATE])
  - [x] Newtypes encapsulate implementation details ([C-NEWTYPE-HIDE])
- **Necessities** *(to whom they matter, they really matter)*
  - [x] Public dependencies of a stable crate are stable ([C-STABLE])
  - [x] Crate and its dependencies have a permissive license ([C-PERMISSIVE])

[C-REEXPORT]: https://rust-lang-nursery.github.io/api-guidelines/organization.html#c-reexport
[C-HIERARCHY]: https://rust-lang-nursery.github.io/api-guidelines/organization.html#c-hierarchy

[C-CASE]: https://rust-lang-nursery.github.io/api-guidelines/naming.html#c-case
[C-CONV]: https://rust-lang-nursery.github.io/api-guidelines/naming.html#c-naming
[C-ITER]: https://rust-lang-nursery.github.io/api-guidelines/naming.html#c-iter
[C-ITER-TY]: https://rust-lang-nursery.github.io/api-guidelines/naming.html#c-iter-ty
[C-OWN-SUFFIX]: https://rust-lang-nursery.github.io/api-guidelines/naming.html#c-own-suffix
[C-GETTERS]: https://rust-lang-nursery.github.io/api-guidelines/naming.html#c-getters
[C-FEATURE]: https://rust-lang-nursery.github.io/api-guidelines/naming.html#c-feature

[C-COMMON-TRAITS]: https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-common-traits
[C-CONV-TRAITS]: https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-conv-traits
[C-COLLECT]: https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-collect
[C-SERDE]: https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-serde
[C-SERDE-CFG]: https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-serde-cfg
[C-SEND-SYNC]: https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-send-sync
[C-SEND-SYNC-ERR]: https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-send-sync-err
[C-MEANINGFUL-ERR]: https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-meaningful-err
[C-NUM-FMT]: https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-num-fmt

[C-EVOCATIVE]: https://rust-lang-nursery.github.io/api-guidelines/macros.html#c-evocative
[C-MACRO-ATTR]: https://rust-lang-nursery.github.io/api-guidelines/macros.html#c-macro-attr
[C-ANYWHERE]: https://rust-lang-nursery.github.io/api-guidelines/macros.html#c-anywhere
[C-MACRO-VIS]: https://rust-lang-nursery.github.io/api-guidelines/macros.html#c-macro-vis
[C-MACRO-TY]: https://rust-lang-nursery.github.io/api-guidelines/macros.html#c-macro-ty

[C-CRATE-DOC]: https://rust-lang-nursery.github.io/api-guidelines/documentation.html#c-crate-doc
[C-EXAMPLE]: https://rust-lang-nursery.github.io/api-guidelines/documentation.html#c-example
[C-QUESTION-MARK]: https://rust-lang-nursery.github.io/api-guidelines/documentation.html#c-question-mark
[C-ERROR-DOC]: https://rust-lang-nursery.github.io/api-guidelines/documentation.html#c-error-doc
[C-PANIC-DOC]: https://rust-lang-nursery.github.io/api-guidelines/documentation.html#c-panic-doc
[C-LINK]: https://rust-lang-nursery.github.io/api-guidelines/documentation.html#c-link
[C-METADATA]: https://rust-lang-nursery.github.io/api-guidelines/documentation.html#c-metadata
[C-DOCS-RS]: https://rust-lang-nursery.github.io/api-guidelines/documentation.html#c-docs-rs
[C-HTML-ROOT]: https://rust-lang-nursery.github.io/api-guidelines/documentation.html#c-html-root
[C-RELNOTES]: https://rust-lang-nursery.github.io/api-guidelines/documentation.html#c-relnotes

[C-SMART-PTR]: https://rust-lang-nursery.github.io/api-guidelines/predictability.html#c-smart-ptr
[C-CONV-SPECIFIC]: https://rust-lang-nursery.github.io/api-guidelines/predictability.html#c-conv-specific
[C-METHOD]: https://rust-lang-nursery.github.io/api-guidelines/predictability.html#c-method
[C-NO-OUT]: https://rust-lang-nursery.github.io/api-guidelines/predictability.html#c-no-out
[C-OVERLOAD]: https://rust-lang-nursery.github.io/api-guidelines/predictability.html#c-overload
[C-DEREF]: https://rust-lang-nursery.github.io/api-guidelines/predictability.html#c-deref
[C-DEREF-FAIL]: https://rust-lang-nursery.github.io/api-guidelines/predictability.html#c-deref-fail
[C-CTOR]: https://rust-lang-nursery.github.io/api-guidelines/predictability.html#c-ctor

[C-INTERMEDIATE]: https://rust-lang-nursery.github.io/api-guidelines/flexibility.html#c-intermediate
[C-CALLER-CONTROL]: https://rust-lang-nursery.github.io/api-guidelines/flexibility.html#c-caller-control
[C-GENERIC]: https://rust-lang-nursery.github.io/api-guidelines/flexibility.html#c-generic
[C-OBJECT]: https://rust-lang-nursery.github.io/api-guidelines/flexibility.html#c-object

[C-NEWTYPE]: https://rust-lang-nursery.github.io/api-guidelines/type-safety.html#c-newtype
[C-CUSTOM-TYPE]: https://rust-lang-nursery.github.io/api-guidelines/type-safety.html#c-custom-type
[C-BITFLAG]: https://rust-lang-nursery.github.io/api-guidelines/type-safety.html#c-bitflag
[C-BUILDER]: https://rust-lang-nursery.github.io/api-guidelines/type-safety.html#c-builder

[C-VALIDATE]: https://rust-lang-nursery.github.io/api-guidelines/dependability.html#c-validate
[C-DTOR-FAIL]: https://rust-lang-nursery.github.io/api-guidelines/dependability.html#c-dtor-fail
[C-DTOR-BLOCK]: https://rust-lang-nursery.github.io/api-guidelines/dependability.html#c-dtor-block

[C-DEBUG]: https://rust-lang-nursery.github.io/api-guidelines/debuggability.html#c-debug
[C-DEBUG-NONEMPTY]: https://rust-lang-nursery.github.io/api-guidelines/debuggability.html#c-debug-nonempty

[C-STRUCT-PRIVATE]: https://rust-lang-nursery.github.io/api-guidelines/future-proofing.html#c-struct-private
[C-NEWTYPE-HIDE]: https://rust-lang-nursery.github.io/api-guidelines/future-proofing.html#c-newtype-hide

[C-STABLE]: https://rust-lang-nursery.github.io/api-guidelines/necessities.html#c-stable
[C-PERMISSIVE]: https://rust-lang-nursery.github.io/api-guidelines/necessities.html#c-permissive

### PR DESCRIPTION
**Pull request by notriddle**
Part of #31: C-LINK

Fixes #31, since C-LINK is the last open checkmark on that mega-issue.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -133,8 +133,8 @@ pub fn clean(src: &str) -> String {
 ///    # }
 ///    ```
 ///
-///  * the `class` attribute is in `allowed_classes` and is in the
-///    correspinding `tag_attributes` or in `generic_attributes`
+///  * the `class` attribute is in [`allowed_classes`] and is in the
+///    correspinding [`tag_attributes`] or in [`generic_attributes`]
```
Comment (lnicola) [inline src/lib.rs]: corresponding

File: src/lib.rs [hunk 2]
```diff
@@ -424,7 +432,7 @@ impl<'a> Builder<'a> {
         self
     }
 
-    /// Constructs an `Ammonia` instance configured with the default options.
+    /// Constructs a [`Builder`] instance configured with the [default options].
```
Comment (lnicola) [inline src/lib.rs]: Good catch

**Comment by lnicola** [comment]
I've fixed the typo and sneaked in a couple of empty lines to match the other examples.

**Comment by notriddle** [comment]
bors r= @lnicola , @notriddle

**Comment by notriddle** [comment]
bors r=@lnicola,@notriddle

Build flaked out.

**Comment by notriddle** [comment]
@lnicola r+ it when you're done. 👍

**Pull request merged**